quandl/model/merged_dataset.py

```diff
@@ -154,8 +103,76 @@ def _in_query_param(self, name, **options):
                 name in options['params'])
 
     def __getattr__(self, k):
-        if k[0] == '_':
+        if k[0] == '_' and k != '_raw_data':
             raise AttributeError(k)
-        elif k in self._datasets[0].raw_data:
+        elif k in self.__dataset__()[0].__get_raw_data__():
             return self._get_dataset_attribute(k)
         return super(MergedDataset, self).__getattr__(k)
+
+    def __get_raw_data__(self):
+        self.__dataset__()
+        return ModelBase.__get_raw_data__(self)
+
+    def __dataset__(self):
+        if self._datasets:
+            return self._datasets
+
+        if not isinstance(self.dataset_codes, list):
+            raise ValueError('dataset codes must be specified in a list')
+        # column_index is handled by individual dataset get's
+        if 'params' in self.options:
+            self.options['params'].pop("column_index", None)
+        self._datasets = list([self._get_dataset(dataset_code, **self.options)
+                               for dataset_code in self.dataset_codes])
+
+        # raw data will be initialized internally
+        self._initialize_raw_data(self._datasets)
+
+        return self._datasets
+
+    def _initialize_raw_data(self, datasets):
+        self._raw_data = {}
+        if not datasets:
+            return self._raw_data
+        self.raw_data = datasets[0].__get_raw_data__().copy()
```

Comment:
we are mixing `_raw_data` and `raw_data`, what's the difference between the two?